fence/agents/powerman/fence_powerman.py

```diff
@@ -0,0 +1,286 @@
+		cmd = ["-q","--exprange"]
+		try:
+			result, ret_code = self._run(cmd, False)
+		except OSError as e:
+			logging.error("PowerMan Error: The command '%s' failed: %s\n", cmd, e)
+			return -1
+		except ValueError as e:
+			logging.error("PowerMan Error: Popen: invalid arguments: %s\n", e)
+			return -1
+		if ret_code < 0:
+			# there was an error with the command
+			return ret_code
+		else:
+			state = {}
+			for line in result[0].split('\n'):
+				if len(line) > 2:
+					fields = line.split(':')
+					if len(fields) == 2:
+						state[fields[0]] = (fields[0],fields[1])
+			return state
+
+	def query(self, host):
+		cmd = ["--query", host]
+		try:
+			result, ret_code = self._run(cmd, True)
+		except OSError as e:
+			logging.error("PowerMan Error: The command '%s' failed: %s\n", cmd, e)
+			return -1
+		except ValueError as e:
+			logging.error("PowerMan Error: Popen: invalid arguments: %s\n", e)
+			return -1
+		if ret_code < 0:
+			# there was an error with the command
+			return ret_code
+		else:
+			res = result.split('\n')
+			res = [r.split() for r in res]
+			# find the host in command's returned output
+			for lst in res:
+				if lst[0] == 'No' and lst[1] == 'such' and lst[2] == 'nodes:':
+					return -1
+				if host in lst:
+					return lst[0][:-1] # lst[0] would be 'off:'-- this removes the colon
+			# host isn't in the output
+			return -1
+
+
+def get_power_status(conn, options):
+	logging.debug("get_power_status function:\noptions: %s\n", str(options))
+	pm = PowerMan(options['--powerman-path'], options['--ip'], options['--ipport'])
+	# if Pacemaker is checking the status of the Powerman server...
+	if options['--action'] == 'monitor':
+		if pm.is_running():
+			logging.debug("Powerman is running\n")
+			return "on"
+		logging.debug("Powerman is NOT running\n")
+		return "error"
+	else:
+		status = pm.query(options['--plug'])
+		if isinstance(int, type(status)):
+			# query only returns ints on error
+			logging.error("get_power_status: query returned %s\n", str(status))
+			fail(EC_STATUS)
+		return status
+
+
+def set_power_status(conn, options):
+	logging.debug("set_power_status function:\noptions: %s", str(options))
+	pm = PowerMan(options['--powerman_path'], options['--ip'], options['--ipport'])
+
+	action = options["--action"]
+	if action == "on":
+		pm.on(options['--plug'])
+	elif action == "off":
+		pm.off(options['--plug'])
+
+	return
+
+
+def get_list(conn, options):
+	logging.debug("get_list function:\noptions: %s", str(options))
+	pm = PowerMan(options['--powerman_path'], options['--ip'], options['--ipport'])
+
+	outlets = pm.list()
+	logging.debug("get_list outlets.keys: %s", str(outlets.keys()))
+	return outlets
+
+
+def define_new_opts():
+	all_opt["powerman_path"] = {
+		"getopt" : ":",
+		"longopt" : "powerman-path",
+		"help" : "--powerman-path=[path]         Path to powerman binary",
+		"required" : "0",
+		"shortdesc" : "Path to powerman binary",
+		"default" : "/usr/bin/powerman",
+		"order": 200
+	}
```

Comment:
Please chnage default value to "@POWERMAN_PATH@" and add line to configure.in (line ~185 at root dir) and to make/fencebuild.mk 

This way the default can be distribution dependant.